src/frontend/components/SelectContent/index.tsx

```diff
@@ -0,0 +1,430 @@
+  addMessage: MessageDescriptor;
+  newTitle: MessageDescriptor;
+  newLtiUrl: string;
+  emptyMessage: MessageDescriptor;
+  items: Nullable<[Video | Document]>;
+  selectUrl: any;
+}) => {
+  const intl = useIntl();
+  const size = React.useContext(ResponsiveContext);
+
+  return (
+    <React.Fragment>
+      <Box direction="row">
+        <Heading level="2">{title}</Heading>
+        <Button
+          color={'brand'}
+          label={<FormattedMessage {...addMessage} />}
+          onClick={() => selectUrl(newLtiUrl, intl.formatMessage(newTitle))}
+        />
+      </Box>
+
+      {items && items.length > 0 ? (
+        <Grid
+          columns={size !== 'small' ? 'medium' : '100%'}
+          gap="medium"
+          rows="medium"
+        >
+          {items?.map((item, index) => (
+            <ContentCard
+              content={item}
+              key={index}
+              onClick={() => selectUrl(item!.lti_url, item!.title)}
+            />
+          ))}
+        </Grid>
+      ) : (
+        <Box>
+          <FormattedMessage {...emptyMessage} />
+          <Anchor label={<FormattedMessage {...addMessage} />} />
+        </Box>
+      )}
+    </React.Fragment>
+  );
+};
+
+interface SelectContentProps {
+  documents?: [Document];
+  videos?: [Video];
+  new_document_url?: string;
+  new_video_url?: string;
+  lti_select_form_action_url?: string;
+  lti_select_form_data?: {
+    [key: string]: string;
+  };
+  lti_content_selected?: boolean;
+}
+
+const SelectContent = ({
+  documents,
+  videos,
+  new_document_url,
+  new_video_url,
+  lti_select_form_action_url,
+  lti_select_form_data,
+  lti_content_selected,
+}: SelectContentProps) => {
+  const [open, setOpen] = React.useState(false);
+  const [selectedTitle, setSelectedTitle] = React.useState('');
+  const [selectedUrl, setSelectedUrl] = React.useState('');
+  const [contentItemsValue, setContentItemsValue] = React.useState('');
+  const ltiTitle = React.useRef<HTMLInputElement>(null);
+  const formRef = React.useRef<HTMLFormElement>(null);
+
+  function selectUrl(ltiUrl: string, title: string) {
+    setSelectedTitle(title);
+    setSelectedUrl(ltiUrl);
+    setOpen(true);
+  }
+
+  const selectTitle = (event: React.ChangeEvent<HTMLInputElement>) =>
+    setSelectedTitle(event.target!.value);
+
+  function confirmSelect() {
+    setSelectedTitle(ltiTitle.current!.value);
+
+    const contentItems = {
+      '@context': 'http://purl.imsglobal.org/ctx/lti/v1/ContentItem',
+      '@graph': [
+        {
+          '@type': 'ContentItem',
+          url: selectedUrl,
+          title: selectedTitle,
+          frame: [],
+        },
+      ],
+    };
+
+    setContentItemsValue(JSON.stringify(contentItems));
+  }
+
+  function cancelSelect() {
+    setSelectedTitle('');
+    setSelectedUrl('');
+    onClose();
+  }
+
+  const onClose = () => setOpen(false);
+
+  useEffect(() => {
+    if (contentItemsValue || lti_content_selected) {
+      formRef.current!.submit();
```

Comment:
You should test if `formRef.current` is defined first and exist this function if not.